From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: xucy <xucy0609@gmail.com>
Date: Mon, 01 Jan 2026 00:00:00 +0000
Subject: [PATCH] Rebrand (consolidated)

Consolidated from 3 original patches:
  - 001-Rebrand-to-Luminol.patch
  - 015-Rebrand-to-Mili.patch
  - 018-Rename-package-me-earthme-luminol-to-fun-bm-mili-lmili.patch


diff --git a/src/main/java/ca/spottedleaf/moonrise/common/util/TickThread.java b/src/main/java/ca/spottedleaf/moonrise/common/util/TickThread.java
--- a/src/main/java/ca/spottedleaf/moonrise/common/util/TickThread.java
+++ b/src/main/java/ca/spottedleaf/moonrise/common/util/TickThread.java
@@ -48,14 +48,14 @@
      */
     @Deprecated
     public static void ensureTickThread(final String reason) {
-        if (!isTickThread() && !fun.bm.mili.config.modules.experiment.DisableAsyncCatcherConfig.enabled) { // Luminol
+        if (!isTickThread() && !fun.bm.mili.config.modules.experiment.DisableAsyncCatcherConfig.enabled) { // Luminol
             LOGGER.error("Thread failed main thread check: " + reason + ", context=" + getThreadContext(), new Throwable());
             throw new IllegalStateException(reason);
         }
     }
 
     public static void ensureTickThread(final Level world, final BlockPos pos, final String reason) {
-        if (!isTickThreadFor(world, pos) && !fun.bm.mili.config.modules.experiment.DisableAsyncCatcherConfig.enabled) { // Luminol
+        if (!isTickThreadFor(world, pos) && !fun.bm.mili.config.modules.experiment.DisableAsyncCatcherConfig.enabled) { // Luminol
             final String ex = "Thread failed main thread check: " +
                                reason + ", context=" + getThreadContext() + ", world=" + WorldUtil.getWorldName(world) + ", block_pos=" + pos;
             LOGGER.error(ex, new Throwable());
@@ -64,7 +64,7 @@
     }
 
     public static void ensureTickThread(final Level world, final BlockPos pos, final int blockRadius, final String reason) {
-        if (!isTickThreadFor(world, pos, blockRadius) && !fun.bm.mili.config.modules.experiment.DisableAsyncCatcherConfig.enabled) { // Luminol
+        if (!isTickThreadFor(world, pos, blockRadius) && !fun.bm.mili.config.modules.experiment.DisableAsyncCatcherConfig.enabled) { // Luminol
             final String ex = "Thread failed main thread check: " +
                 reason + ", context=" + getThreadContext() + ", world=" + WorldUtil.getWorldName(world) + ", block_pos=" + pos + ", block_radius=" + blockRadius;
             LOGGER.error(ex, new Throwable());
@@ -73,7 +73,7 @@
     }
 
     public static void ensureTickThread(final Level world, final ChunkPos pos, final String reason) {
-        if (!isTickThreadFor(world, pos) && !fun.bm.mili.config.modules.experiment.DisableAsyncCatcherConfig.enabled) { // Luminol
+        if (!isTickThreadFor(world, pos) && !fun.bm.mili.config.modules.experiment.DisableAsyncCatcherConfig.enabled) { // Luminol
             final String ex = "Thread failed main thread check: " +
                 reason + ", context=" + getThreadContext() + ", world=" + WorldUtil.getWorldName(world) + ", chunk_pos=" + pos;
             LOGGER.error(ex, new Throwable());
@@ -82,7 +82,7 @@
     }
 
     public static void ensureTickThread(final Level world, final int chunkX, final int chunkZ, final String reason) {
-        if (!isTickThreadFor(world, chunkX, chunkZ) && !fun.bm.mili.config.modules.experiment.DisableAsyncCatcherConfig.enabled) { // Luminol
+        if (!isTickThreadFor(world, chunkX, chunkZ) && !fun.bm.mili.config.modules.experiment.DisableAsyncCatcherConfig.enabled) { // Luminol
             final String ex = "Thread failed main thread check: " +
                 reason + ", context=" + getThreadContext() + ", world=" + WorldUtil.getWorldName(world) + ", chunk_pos=" + new ChunkPos(chunkX, chunkZ);
             LOGGER.error(ex, new Throwable());
@@ -91,7 +91,7 @@
     }
 
     public static void ensureTickThread(final Entity entity, final String reason) {
-        if (!isTickThreadFor(entity) && !fun.bm.mili.config.modules.experiment.DisableAsyncCatcherConfig.enabled) { // Luminol
+        if (!isTickThreadFor(entity) && !fun.bm.mili.config.modules.experiment.DisableAsyncCatcherConfig.enabled) { // Luminol
             final String ex = "Thread failed main thread check: " +
                 reason + ", context=" + getThreadContext() + ", entity=" + EntityUtil.dumpEntity(entity);
             LOGGER.error(ex, new Throwable());
@@ -100,7 +100,7 @@
     }
 
     public static void ensureTickThread(final Level world, final AABB aabb, final String reason) {
-        if (!isTickThreadFor(world, aabb) && !fun.bm.mili.config.modules.experiment.DisableAsyncCatcherConfig.enabled) { // Luminol
+        if (!isTickThreadFor(world, aabb) && !fun.bm.mili.config.modules.experiment.DisableAsyncCatcherConfig.enabled) { // Luminol
             final String ex = "Thread failed main thread check: " +
                 reason + ", context=" + getThreadContext() + ", world=" + WorldUtil.getWorldName(world) + ", aabb=" + aabb;
             LOGGER.error(ex, new Throwable());
@@ -109,7 +109,7 @@
     }
 
     public static void ensureTickThread(final Level world, final double blockX, final double blockZ, final String reason) {
-        if (!isTickThreadFor(world, blockX, blockZ) && !fun.bm.mili.config.modules.experiment.DisableAsyncCatcherConfig.enabled) { // Luminol
+        if (!isTickThreadFor(world, blockX, blockZ) && !fun.bm.mili.config.modules.experiment.DisableAsyncCatcherConfig.enabled) { // Luminol
             final String ex = "Thread failed main thread check: " +
                 reason + ", context=" + getThreadContext() + ", world=" + WorldUtil.getWorldName(world) + ", block_pos=" + new Vec3(blockX, 0.0, blockZ);
             LOGGER.error(ex, new Throwable());

diff --git a/src/main/java/com/destroystokyo/paper/Metrics.java b/src/main/java/com/destroystokyo/paper/Metrics.java
--- a/src/main/java/com/destroystokyo/paper/Metrics.java
+++ b/src/main/java/com/destroystokyo/paper/Metrics.java
@@ -593,7 +593,7 @@
             boolean logFailedRequests = config.getBoolean("logFailedRequests", false);
             // Only start Metrics, if it's enabled in the config
             if (config.getBoolean("enabled", true)) {
-                Metrics metrics = new Metrics("Folia", serverUUID, logFailedRequests, Bukkit.getLogger()); // Folia - we have our own bstats page
+                Metrics metrics = new Metrics("Luminol", serverUUID, logFailedRequests, Bukkit.getLogger()); // Folia - we have our own bstats page // Luminol
 
                 metrics.addCustomChart(new Metrics.SimplePie("minecraft_version", () -> {
                     String minecraftVersion = Bukkit.getVersion();
@@ -607,11 +607,11 @@
                 final String implVersion = org.bukkit.craftbukkit.Main.class.getPackage().getImplementationVersion();
                 if (implVersion != null) {
                     final String buildOrHash = implVersion.substring(implVersion.lastIndexOf('-') + 1);
-                    paperVersion = "git-Folia-%s-%s".formatted(Bukkit.getServer().getMinecraftVersion(), buildOrHash); // Folia - we have our own bstats page
+                    paperVersion = "git-Luminol-%s-%s".formatted(Bukkit.getServer().getMinecraftVersion(), buildOrHash); // Folia - we have our own bstats page // Luminol - Rebrand to Luminol
                 } else {
                     paperVersion = "unknown";
                 }
-                metrics.addCustomChart(new Metrics.SimplePie("folia_version", () -> paperVersion)); // Folia - we have our own bstats page
+                metrics.addCustomChart(new Metrics.SimplePie("luminol_version", () -> paperVersion)); // Folia - we have our own bstats page // Luminol
 
                 metrics.addCustomChart(new Metrics.DrilldownPie("java_version", () -> {
                     Map<String, Map<String, Integer>> map = new HashMap<>();
@@ -593,7 +593,7 @@
             boolean logFailedRequests = config.getBoolean("logFailedRequests", false);
             // Only start Metrics, if it's enabled in the config
             if (config.getBoolean("enabled", true)) {
-                Metrics metrics = new Metrics("Luminol", serverUUID, logFailedRequests, Bukkit.getLogger()); // Folia - we have our own bstats page // Luminol
+                Metrics metrics = new Metrics("Mili", serverUUID, logFailedRequests, Bukkit.getLogger()); // Folia - we have our own bstats page // Luminol // Lophine // Mili
 
                 metrics.addCustomChart(new Metrics.SimplePie("minecraft_version", () -> {
                     String minecraftVersion = Bukkit.getVersion();
@@ -607,11 +607,11 @@
                 final String implVersion = org.bukkit.craftbukkit.Main.class.getPackage().getImplementationVersion();
                 if (implVersion != null) {
                     final String buildOrHash = implVersion.substring(implVersion.lastIndexOf('-') + 1);
-                    paperVersion = "git-Luminol-%s-%s".formatted(Bukkit.getServer().getMinecraftVersion(), buildOrHash); // Folia - we have our own bstats page // Luminol - Rebrand to Luminol
+                    paperVersion = "git-Lophine-%s-%s".formatted(Bukkit.getServer().getMinecraftVersion(), buildOrHash); // Folia - we have our own bstats page // Luminol - Rebrand to Luminol // Lophine // Mili - Rebrand to Mili
                 } else {
                     paperVersion = "unknown";
                 }
-                metrics.addCustomChart(new Metrics.SimplePie("luminol_version", () -> paperVersion)); // Folia - we have our own bstats page // Luminol
+                metrics.addCustomChart(new Metrics.SimplePie("lophine_version", () -> paperVersion)); // Folia - we have our own bstats page // Luminol // Lophine // Mili
 
                 metrics.addCustomChart(new Metrics.DrilldownPie("java_version", () -> {
                     Map<String, Map<String, Integer>> map = new HashMap<>();

diff --git a/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java b/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java
--- a/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java
+++ b/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java
@@ -40,8 +40,8 @@
     private static final ComponentLogger COMPONENT_LOGGER = ComponentLogger.logger(LogManager.getRootLogger().getName());
     private static final int DISTANCE_ERROR = -1;
     private static final int DISTANCE_UNKNOWN = -2;
-    private static final String DOWNLOAD_PAGE = "https://papermc.io/downloads/folia"; // Folia
-    private static final String REPOSITORY = "PaperMC/Folia"; // Folia
+    private static final String DOWNLOAD_PAGE = "https://github.com/LuminolMC/Luminol"; // Luminol
+    private static final String REPOSITORY = "LuminolMC/Luminol"; // Luminol
     private static final ServerBuildInfo BUILD_INFO = ServerBuildInfo.buildInfo();
     private static final String USER_AGENT = BUILD_INFO.brandName() + "/" + BUILD_INFO.asString(VERSION_SIMPLE) + " (https://papermc.io)";
     private static final Gson GSON = new Gson();
@@ -72,7 +72,7 @@
             COMPONENT_LOGGER.warn(text("*** You are running a development version without access to version information ***"));
         } else {
             final Optional<MinecraftVersionFetcher> apiResult = fetchMinecraftVersionList();
-            if (buildNumber.isPresent()) {
+            if (buildNumber.isPresent() && false) { // Luminol
                 distance = fetchDistanceFromSiteApi(buildNumber.getAsInt());
             } else {
                 final Optional<String> gitBranch = BUILD_INFO.gitBranch();
@@ -40,8 +40,8 @@
     private static final ComponentLogger COMPONENT_LOGGER = ComponentLogger.logger(LogManager.getRootLogger().getName());
     private static final int DISTANCE_ERROR = -1;
     private static final int DISTANCE_UNKNOWN = -2;
-    private static final String DOWNLOAD_PAGE = "https://github.com/LuminolMC/Luminol"; // Luminol
-    private static final String REPOSITORY = "LuminolMC/Luminol"; // Luminol
+    private static final String DOWNLOAD_PAGE = "https://github.com/xucy10/Mili"; // Luminol // Lophine // Mili
+    private static final String REPOSITORY = "xucy10/Mili"; // Luminol // Lophine // Mili
     private static final ServerBuildInfo BUILD_INFO = ServerBuildInfo.buildInfo();
     private static final String USER_AGENT = BUILD_INFO.brandName() + "/" + BUILD_INFO.asString(VERSION_SIMPLE) + " (https://papermc.io)";
     private static final Gson GSON = new Gson();

diff --git a/src/main/java/com/destroystokyo/paper/network/PaperServerListPingEventImpl.java b/src/main/java/com/destroystokyo/paper/network/PaperServerListPingEventImpl.java
--- a/src/main/java/com/destroystokyo/paper/network/PaperServerListPingEventImpl.java
+++ b/src/main/java/com/destroystokyo/paper/network/PaperServerListPingEventImpl.java
@@ -14,7 +14,7 @@
 
     PaperServerListPingEventImpl(MinecraftServer server, StatusClient client, int protocolVersion, @Nullable CachedServerIcon icon) {
         super(client, server.motd(), server.getPlayerCount(), server.getMaxPlayers(),
-                fun.bm.mili.config.modules.misc.ServerModNameConfig.fakeVanilla ? server.getServerVersion() : server.getServerModName() + ' ' + server.getServerVersion(), protocolVersion, icon);//Luminol - Fake vanilla mode
+                fun.bm.mili.config.modules.misc.ServerModNameConfig.fakeVanilla ? server.getServerVersion() : server.getServerModName() + ' ' + server.getServerVersion(), protocolVersion, icon);//Luminol - Fake vanilla mode
         this.server = server;
     }
 

diff --git a/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java b/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java
--- a/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java
@@ -31,6 +31,7 @@
     private static final String ATTRIBUTE_GIT_COMMIT = "Git-Commit";
 
     private static final String BRAND_PAPER_NAME = "Paper";
+    private static final String BRAND_LUMINOL_NAME = "Luminol";
 
     private static final String BUILD_DEV = "DEV";
 
@@ -43,9 +44,9 @@
         this(
             getManifestAttribute(manifest, ATTRIBUTE_BRAND_ID)
                 .map(Key::key)
-                .orElse(Key.key("papermc", "folia")), // Folia
+                .orElse(BRAND_LUMINOL_ID), // Folia // Luminol
             getManifestAttribute(manifest, ATTRIBUTE_BRAND_NAME)
-                .orElse("Folia"), // Folia
+                .orElse(BRAND_LUMINOL_NAME), // Folia // Luminol
             SharedConstants.getCurrentVersion().id(),
             SharedConstants.getCurrentVersion().name(),
             getManifestAttribute(manifest, ATTRIBUTE_BUILD_NUMBER)
@@ -62,7 +63,7 @@
 
     @Override
     public boolean isBrandCompatible(final @NotNull Key brandId) {
-        return brandId.equals(this.brandId);
+        return brandId.equals(this.brandId) || brandId.equals(BRAND_PAPER_ID) || brandId.equals(BRAND_FOLIA_ID); // Luminol
     }
 
     @Override
@@ -32,7 +32,8 @@
     private static final String ATTRIBUTE_GIT_COMMIT = "Git-Commit";
 
     private static final String BRAND_PAPER_NAME = "Paper";
-    private static final String BRAND_LUMINOL_NAME = "Luminol";
+    private static final String BRAND_MILI_NAME = "Mili";
+    private static final Key BRAND_MILI_ID = Key.key("xucy10", "mili"); // Mili
 
     private static final String BUILD_DEV = "DEV";
 
@@ -45,9 +46,9 @@
         this(
             getManifestAttribute(manifest, ATTRIBUTE_BRAND_ID)
                 .map(Key::key)
-                .orElse(BRAND_LUMINOL_ID), // Folia // Luminol
+                .orElse(BRAND_MILI_ID), // Folia // Luminol // Lophine // Mili
             getManifestAttribute(manifest, ATTRIBUTE_BRAND_NAME)
-                .orElse(BRAND_LUMINOL_NAME), // Folia // Luminol
+                .orElse(BRAND_MILI_NAME), // Folia // Luminol // Lophine // Mili
             SharedConstants.getCurrentVersion().id(),
             SharedConstants.getCurrentVersion().name(),
             getManifestAttribute(manifest, ATTRIBUTE_BUILD_NUMBER)
@@ -64,7 +65,7 @@
 
     @Override
     public boolean isBrandCompatible(final @NotNull Key brandId) {
-        return brandId.equals(this.brandId) || brandId.equals(BRAND_PAPER_ID) || brandId.equals(BRAND_FOLIA_ID); // Luminol
+        return brandId.equals(this.brandId) || brandId.equals(BRAND_PAPER_ID) || brandId.equals(BRAND_FOLIA_ID) || brandId.equals(BRAND_MILI_ID); // Folia // Luminol // Lophine // Mili
     }
 
     @Override

diff --git a/src/main/java/io/papermc/paper/plugin/provider/type/paper/PaperPluginProviderFactory.java b/src/main/java/io/papermc/paper/plugin/provider/type/paper/PaperPluginProviderFactory.java
--- a/src/main/java/io/papermc/paper/plugin/provider/type/paper/PaperPluginProviderFactory.java
+++ b/src/main/java/io/papermc/paper/plugin/provider/type/paper/PaperPluginProviderFactory.java
@@ -25,7 +25,7 @@
     @Override
     public PaperPluginParent build(JarFile file, PaperPluginMeta configuration, Path source) {
         // Folia start - block plugins not marked as supported
-        if (!configuration.isFoliaSupported() && !fun.bm.mili.config.modules.unsupported.DisableCheckForFoliaSupported.disableForPaper) { // Luminol - disable check for folia supported
+        if (!configuration.isFoliaSupported() && !fun.bm.mili.config.modules.unsupported.DisableCheckForFoliaSupported.disableForPaper) { // Luminol - disable check for folia supported
             throw new RuntimeException("Could not load plugin '" + configuration.getDisplayName() + "' as it is not marked as supporting Folia!");
         }
         // Folia end - block plugins not marked as supported

diff --git a/src/main/java/io/papermc/paper/plugin/provider/type/spigot/SpigotPluginProviderFactory.java b/src/main/java/io/papermc/paper/plugin/provider/type/spigot/SpigotPluginProviderFactory.java
--- a/src/main/java/io/papermc/paper/plugin/provider/type/spigot/SpigotPluginProviderFactory.java
+++ b/src/main/java/io/papermc/paper/plugin/provider/type/spigot/SpigotPluginProviderFactory.java
@@ -27,7 +27,7 @@
     @Override
     public SpigotPluginProvider build(JarFile file, PluginDescriptionFile configuration, Path source) throws InvalidDescriptionException {
         // Folia start - block plugins not marked as supported
-        if (!configuration.isFoliaSupported() && !fun.bm.mili.config.modules.unsupported.DisableCheckForFoliaSupported.disableForPaper) { // Luminol - disable check for folia supported
+        if (!configuration.isFoliaSupported() && !fun.bm.mili.config.modules.unsupported.DisableCheckForFoliaSupported.disableForPaper) { // Luminol - disable check for folia supported
             throw new RuntimeException("Could not load plugin '" + configuration.getDisplayName() + "' as it is not marked as supporting Folia!");
         }
         // Folia end - block plugins not marked as supported

diff --git a/src/main/java/org/bukkit/craftbukkit/CraftChunk.java b/src/main/java/org/bukkit/craftbukkit/CraftChunk.java
--- a/src/main/java/org/bukkit/craftbukkit/CraftChunk.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftChunk.java
@@ -205,7 +205,7 @@
     public boolean isSlimeChunk() {
         // 987234911L is taken from Slime when seeing if a slime can spawn in a chunk
         // Leaf start - Matter - Secure Seed
-        boolean isSlimeChunk = fun.bm.mili.config.modules.function.SecureSeedConfig.enabled
+        boolean isSlimeChunk = fun.bm.mili.config.modules.function.SecureSeedConfig.enabled
             ? level.getChunk(this.getX(), this.getZ()).isSlimeChunk()
             : WorldgenRandom.seedSlimeChunk(this.getX(), this.getZ(), this.getWorld().getSeed(), level.spigotConfig.slimeSeed).nextInt(10) == 0; // Paper
         return this.level.paperConfig().entities.spawning.allChunksAreSlimeChunks || isSlimeChunk;

diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1264,7 +1264,7 @@
         RegistryAccess contextRegistryAccess = registryAccess;
         if (worldGenSettings == null) {
             // Leaf start - Matter - Secure Seed
-            WorldOptions worldOptions = fun.bm.mili.config.modules.function.SecureSeedConfig.enabled
+            WorldOptions worldOptions = fun.bm.mili.config.modules.function.SecureSeedConfig.enabled
                 ? new WorldOptions(creator.seed(), su.plo.matter.Globals.createRandomWorldSeed(), creator.generateStructures(), false)
                 : new WorldOptions(creator.seed(), creator.generateStructures(), creator.bonusChest());
             // Leaf end - Matter - Secure Seed

diff --git a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
--- a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
@@ -2092,7 +2092,7 @@
 
     // Luminol start - Tick regions api
     @Override
-    public fun.bm.mili.lmili.api.ThreadedRegionizer getThreadedRegionizer() {
+    public fun.bm.mili.lmili.api.ThreadedRegionizer getThreadedRegionizer() {
         return this.world.regioniser.threadedRegionizerAPI;
     }
     // Luminol end

diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
@@ -1152,7 +1152,7 @@
     }
 
private void updateWaypoint() {
-        fun.bm.mili.lmili.utils.FoliaServerWaypointManager manager = ((ServerLevel) getHandle().level()).getWaypointManager(); // Luminol - Restore waypoints
+        ServerWaypointManager manager = ((ServerLevel) getHandle().level()).getWaypointManager();
        manager.untrackWaypoint(getHandle());
        manager.trackWaypoint(getHandle());
    }

diff --git a/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java b/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
--- a/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
+++ b/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
@@ -393,7 +393,7 @@
         }
 
         // Folia start - block plugins not marked as supported
-        if (!descriptionFile.isFoliaSupported() && !fun.bm.mili.config.modules.unsupported.DisableCheckForFoliaSupported.disableForPaper) { // Luminol - disable check for folia supported
+        if (!descriptionFile.isFoliaSupported() && !fun.bm.mili.config.modules.unsupported.DisableCheckForFoliaSupported.disableForPaper) { // Luminol - disable check for folia supported
             throw new InvalidPluginException("Plugin " + descriptionFile.getFullName() + " is not marked as supporting regionised multithreading");
         }
         // Folia end - block plugins not marked as supported

diff --git a/src/main/java/org/spigotmc/AsyncCatcher.java b/src/main/java/org/spigotmc/AsyncCatcher.java
--- a/src/main/java/org/spigotmc/AsyncCatcher.java
+++ b/src/main/java/org/spigotmc/AsyncCatcher.java
@@ -5,7 +5,7 @@
 public class AsyncCatcher {
 
     public static void catchOp(String reason) {
-        if (!ca.spottedleaf.moonrise.common.util.TickThread.isTickThread() && !fun.bm.mili.config.modules.experiment.DisableAsyncCatcherConfig.enabled) { // Paper - chunk system  // Luminol
+        if (!ca.spottedleaf.moonrise.common.util.TickThread.isTickThread() && !fun.bm.mili.config.modules.experiment.DisableAsyncCatcherConfig.enabled) { // Paper - chunk system  // Luminol
             MinecraftServer.LOGGER.error("Thread {} failed main thread check: {}", Thread.currentThread().getName(), reason, new Throwable()); // Paper
             throw new IllegalStateException("Asynchronous " + reason + "!");
         }
-- 

diff --git a/src/main/java/org/spigotmc/WatchdogThread.java b/src/main/java/org/spigotmc/WatchdogThread.java
--- a/src/main/java/org/spigotmc/WatchdogThread.java
+++ b/src/main/java/org/spigotmc/WatchdogThread.java
@@ -80,9 +80,10 @@
                     logger.log(Level.SEVERE, "\t *Especially* if it looks like HTTP or MySQL operations are occurring");
                     logger.log(Level.SEVERE, "If you see a world save or edit, then it means you did far more than your server can handle at once");
                     logger.log(Level.SEVERE, "\t If this is the case, consider increasing timeout-time in spigot.yml but note that this will replace the crash with LARGE lag spikes");
-                    logger.log(Level.SEVERE, "If you are unsure or still think this is a Paper bug, please report this to https://github.com/PaperMC/Paper/issues");
+                    logger.log(Level.SEVERE, "DO NOT REPORT THIS TO PAPER!");
+                    logger.log(Level.SEVERE, "If you are unsure or still think this is a Luminol bug, please report this to https://github.com/LuminolMC/Luminol/issues");
                     logger.log(Level.SEVERE, "Be sure to include ALL relevant console errors and Minecraft crash reports");
-                    logger.log(Level.SEVERE, "Paper version: " + Bukkit.getServer().getVersion());
+                    logger.log(Level.SEVERE, "Server version: " + Bukkit.getServer().getVersion());
 
                     if (net.minecraft.world.level.Level.lastPhysicsProblem != null) {
                         logger.log(Level.SEVERE, "------------------------------");

